core/trino-main/src/main/java/io/trino/connector/system/jdbc/FilterUtil.java

```diff
@@ -50,6 +50,20 @@ public static QualifiedTablePrefix tablePrefix(String catalog, Optional<String>
         return new QualifiedTablePrefix(catalog);
     }
 
+    public static Optional<QualifiedTablePrefix> tablePrefixIfPossible(String catalog, Optional<String> schema, Optional<String> table)
```

Comment:
Looking at this & above method, it's unclear what is the difference. 
You need to check the implementation to understand that. Better naming?

Reply:
method is gone in current iteration